pkg/cloud/common/interfaces.go

```diff
@@ -0,0 +1,13 @@
+package common
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CloudProviderAssets represent cloud provider specific assets object. Such as Azure/Aws/OpenStack/etc.
+// Intended to hide provider specific resource management, manifests checks and rendering behind this interface.
+type CloudProviderAssets interface {
+	// GetRenderedResources intended to return ready to apply list of resources.
+	// Expected that resources returned by this method is completely prepared and do not need additional handling.
+	GetRenderedResources() ([]client.Object, error)
```

Comment:
I still think that, until we have a reason to do otherwise, this should be. There's no need to keep the error in case we use it in the future, we can add it back if we need it ever
```suggestion
	GetRenderedResources() []client.Object
```

Reply:
Fair point at the moment. I expect that it _may_ be little bit problematic to get it back in future, but  agree, at this point this err is not needed. Removed in https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/110/commits/5c1cb86916a79392d22b6a64392b3d2c26c5d14d